src/util/datablock/datablock.h

```diff
@@ -12,13 +12,30 @@
 #include "../block.h"
 #include "./datablock_iterator.h"
 
-#define DELETED_MARKER 0xFF
-
 typedef void (*fpDestructor)(void *);
 
 // Number of items in a block. Should always be a power of 2.
 #define DATABLOCK_BLOCK_CAP 16384
 
+// Returns the item header size.
+#define ITEM_HEADER_SIZE (sizeof(DataBlockItemHeader))
+
+// DataBlock item is stored as ||header|data||. This macro retrive the data pointer out of the header pointer.
+#define ITEM_DATA(header) ((void *)(header? header + ITEM_HEADER_SIZE : header))
```

Comment:
I don't think we should try and handle the NULL case.

Reply:
removed